kaldb/src/main/java/com/slack/kaldb/chunk/ReadWriteChunk.java

```diff
@@ -25,65 +25,67 @@
 import java.util.Collection;
 import org.apache.lucene.index.IndexCommit;
 import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
- * A ReadWriteChunkImpl provides a concrete implementation for a shard to which we can write and
- * read the messages we wrote. It provides a unified interface of a shard abstracting the details of
- * the underlying storage implementation.
+ * An ReadWriteChunk provides a base implementation for a shard to which we can write and read the
+ * messages we wrote. It provides a unified interface of a shard abstracting the details of the
+ * underlying storage implementation. There will be 2 implementation for this class, one in the
+ * indexer and one in recovery process. The code that's common for both of these classes will reside
+ * in the base class.
  *
- * <p>Chunk maintains its metadata in the chunkInfo object. The chunkInfo tracks all the info needed
- * for constructing a snapshot.
+ * <p>A ReadWriteChunk maintains its metadata in the chunkInfo object. For example, the data in the
+ * chunkInfo object can be used when publishing a snapshot from the chunk.
  *
- * <p>A ReadWriteChunk goes through the following life cycle.
+ * <p>A ReadWriteChunk goes through the following life cycle. The hooks into the life cycle are
+ * implemented as abstract base methods so derived classes can take custom action on those stages.
  *
  * <p>When a chunk is created it is open for both reads and writes. Since a ReadWriteChunk is
- * ingesting live data, a cluster manager doesn't manage it. Instead, when a chunk in created, it
- * registers a live snapshot and a live search node.
+ * ingesting live data, a cluster manager doesn't manage it. The register method is called after the
+ * chunk is created and deRegister method is called just before chunk close.
```

Comment:
These methods should be named for when they occur, and not what the current implementation does today. For example `afterChunkCreation` and `beforeChunkClose` is better than `register` and `deRegister`, as registration is a concept specific to `IndexingReadWriteChunkImpl`. Future `ReadWriteChunks` will want to take actions at these lifecycle stages that have nothing to do with registration.

This also makes these two methods consistent with `postSnapshot` which already follows this pattern.

Reply:
Good call. Reamed to `postCreate` and `preClose` to be consistent with the snapshot names.